docs/src/proposals/interest-bearing-tokens.md

```diff
@@ -0,0 +1,414 @@
+* call `InitializeHolding` on the token program
+
+### Web3 / Wallets
+
+To properly support i-tokens, wallets and applications will need to make the
+following changes:
+
+* validate program: using the token-list, check that the given program id is legit
+* get holdings: query for holdings owned by the public key, for all official
+token programs listed in the registry (not just `Tokenkeg...`)
+* get balances: avoid deserializing holding data, and instead use
+`getBalance` from RPC to get UI amounts
+* transfer tokens: only use `ApproveChecked` and `TransferChecked` on the token
+program, and always convert UI amount to amount before calling instructions,
+using `UiAmountToAmount`
+* create holding: avoid directly creating an account and initializing,
+and use the associated token account program instead
+
+### On-chain Programs
+
+To properly support i-tokens on-chain, programs will need to make these changes:
+
+* multiple token programs: all programs that support SPL Tokens must always
+include the program id of the token(s) holdings they're passing in.
+For example, token-swap needs to accept separate token programs for token A and B
+* hard-coded token program: avoid using `Tokenkeg...` directly, and delete `id()`
+the SPL token code
+* get holding data: instead of deserializing the account data into an SPL holding, use
+the new wrapper to directly deserialize if the i-token program is "valid"
+* get mint data: same as above
+* UI amount vs token amount: know when the program should use amounts, which
+could be share amounts, and when to use UI amounts
+* transfer: only use `TransferChecked`, which requires passing mints
+
+### Ledger
+
+Ledger integrates with token-list in order to provide verified token program ids
+and token tickers to the Ledger SDK and eliminate the need to store static
+information in the Ledger Solana App.
+
+A wallet or CLI with Ledger integration fetches the token-list and signing
+information. When the user wishes to transfer tokens, the wallet first builds an
+on-device token registry by pushing signed entries from the token-list for the subset
+of tokens referenced by the transaction.
+
+Ledger host applications provide token-list entry metadata is pushed to the device
+via a new Ledger APDU command accepting **TBD_DATA_FORMAT** signed by the
+token-list authority, whose public key is embedded in the Ledger app binary. Entries
+whose signature verification passes are stored in a registry on the device that is
+then referenced by a subsequent SIGN_SOLANA_TRANSACTION APDU command.
+
+## Other New Token Programs
+
+For awhile now, people have been coming up with ideas that likely require a new
+token program to implement. These include:
+
+### Voting Tokens
+
+Outlined in a [GitHub issue](https://github.com/solana-labs/solana-program-library/issues/131),
+voting tokens use additional data about how long a token has been held.
```

Comment:
The opposite logic would be to make it possible to lock tokens for certain amount of time to know for how long they would be held. This would allow to calculate voting power based on the long term commitment (skin in the game).
Users don't really like to deposit governance tokens to vote. It makes them very uncomfortable especially when the governance tokens they posses have substantial economic value. 
A better solution would be to allow the tokens to be locked to gain voting power while still being in user's possession.